Fix k8s podCIDRs for vagrant deployment

romanspb80 opened the pull request:
This patch makes podCIDRs of k8s Node and CiliumNode consistent: vagrant@k8s1:~/go/src/github.com/cilium/cilium$ kubectl get nodes k8s1 -o yaml apiVersion: v1
kind: Node
metadata:
...
spec:
  podCIDR: 10.11.0.0/24
  podCIDRs:
  - 10.11.0.0/24
  - fd04::/112 ...

vagrant@k8s1:~/go/src/github.com/cilium/cilium$ kubectl get ciliumnodes.cilium.io -o yaml apiVersion: v1
items:
- apiVersion: cilium.io/v2 kind: CiliumNode ... spec: ... ipam: podCIDRs: - 10.11.0.0/24 - fd04::/112 ...

Fixes: #19594

Signed-off-by: Roman Ptitcyn <romanspb@yahoo.com>

christarazi commented on pkg/ip/ip_test.go:
```diff
@@ -664,7 +664,7 @@ func (s *IPTestSuite) TestPartitionCIDR(c *C) {
 		createIPNet("10.64.0.0", 10, int(ipv4BitLen)),
 		createIPNet("10.32.0.0", 11, int(ipv4BitLen)),
-		createIPNet("10.16.0.0", 12, int(ipv4BitLen)),
+		createIPNet("10.11.0.0", 24, int(ipv4BitLen)),
```
This seems like an unrelated change. This is test code and this delta is changing one of the test cases. Can you drop this change?

christarazi commented on pkg/labels/cidr/cidr_test.go:
```diff
@@ -177,7 +177,7 @@ func (s *CIDRLabelsSuite) TestIPStringToLabel(c *C) {
 	for _, cidr := range []netip.Prefix{
 		netip.MustParsePrefix("0.0.0.0/0"),
-		netip.MustParsePrefix("10.16.0.0/16"),
+		netip.MustParsePrefix("10.11.0.0/24"),
```
Same here

christarazi commented on pkg/option/config.go:
```diff
@@ -175,7 +175,7 @@ const (
 	FixedIdentityMapping = "fixed-identity-mapping"
 
-	// IPv4Range is the per-node IPv4 endpoint prefix, e.g. 10.16.0.0/16
+	// IPv4Range is the per-node IPv4 endpoint prefix, e.g. 10.11.0.0/24
```
Same here

christarazi commented on pkg/policy/cidr_test.go:
```diff
@@ -111,7 +111,7 @@ func (ds *PolicyTestSuite) TestGetCIDRPrefixes(c *C) {
 		"10.64.0.0/10",
 		"10.32.0.0/11",
-		"10.16.0.0/12",
+		"10.11.0.0/24",
```
Same here

romanspb80 replied:
Thanks Chris
Done

christarazi reviewed (changes requested):
Thanks for the PR. I think the changes need only be scoped to `contrib/vagrant`.

julianwiedmann commented on contrib/vagrant/scripts/helpers.bash:
```diff
@@ -56,10 +56,10 @@ kubernetes_master="${controllers_ips[0]}"
-k8s_node_cidr_v4_mask_size=${K8S_NODE_CIDR_MASK_SIZE:-"16"} # 1st Node: 10.16.0.1-10.16.255.254, 2nd Node: 10.17.0.1-10.17.255.254...
-k8s_node_cidr_v6_mask_size=${K8S_NODE_CIDR_V6_MASK_SIZE:-"112"} # 1st Node: 10.16.0.1-10.16.255.254, 2nd Node: 10.17.0.1-10.17.255.254...
+k8s_node_cidr_v4_mask_size=${K8S_NODE_CIDR_MASK_SIZE:-"24"} # 1st Node: 10.11.0.1-10.11.0.255, 2nd Node: 10.11.1.1-10.11.1.255...
+k8s_node_cidr_v6_mask_size=${K8S_NODE_CIDR_V6_MASK_SIZE:-"112"} # 1st Node: 10.11.0.1-10.11.0.255, 2nd Node: 10.11.1.1-10.11.1.255...
```
This doesn't seem correct - now each per-node Pod CIDR would have the same size as the cluster's total Pod CIDR.

Did the CIDR sizes actually need fixing?

romanspb80 replied:
> This doesn't seem correct - now each per-node Pod CIDR would have the same size as the cluster's total Pod CIDR.
> 
> Did the CIDR sizes actually need fixing?

Thanks for the review.
I have reduced the mask value "K8S_CLUSTER_CIDR" by 4 bits

julianwiedmann commented on contrib/vagrant/scripts/helpers.bash:
```diff
@@ -56,10 +56,10 @@ kubernetes_master="${controllers_ips[0]}"
-k8s_node_cidr_v4_mask_size=${K8S_NODE_CIDR_MASK_SIZE:-"16"} # 1st Node: 10.16.0.1-10.16.255.254, 2nd Node: 10.17.0.1-10.17.255.254...
-k8s_node_cidr_v6_mask_size=${K8S_NODE_CIDR_V6_MASK_SIZE:-"112"} # 1st Node: 10.16.0.1-10.16.255.254, 2nd Node: 10.17.0.1-10.17.255.254...
+k8s_node_cidr_v4_mask_size=${K8S_NODE_CIDR_MASK_SIZE:-"24"} # 1st Node: 10.11.0.1-10.11.0.255, 2nd Node: 10.11.1.1-10.11.1.255...
+k8s_node_cidr_v6_mask_size=${K8S_NODE_CIDR_V6_MASK_SIZE:-"112"} # 1st Node: 10.11.0.1-10.11.0.255, 2nd Node: 10.11.1.1-10.11.1.255...
```
This still looks odd - having a cluster-wide IPv6 CIDR of `FD04::/112`, the per-node CIDR mask can't be `112`.

romanspb80 replied:
Hi @julianwiedmann . Thanks for the comments. I made corrections. Could you please check?

julianwiedmann reviewed (changes requested):
Hi @romanspb80 , thanks! Besides the IPv6 inconsistency in the change (as noted below), a more general remark - we want these CIDRs to end up looking identical to what gets passed to https://github.com/cilium/cilium/blob/80af06ee3c224f529961cd5ac546de27516ccb10/contrib/vagrant/start.sh#L332-L333

Right now the per-node CIDR size gets auto-sized there, but it can be configured with `--cluster-pool-ipv4-mask-size`.

romanspb80 committed:
Fix k8s podCIDRs for vagrant deployment

This patch makes podCIDRs of k8s Node and CiliumNode consistent:
vagrant@k8s1:~/go/src/github.com/cilium/cilium$ kubectl get nodes k8s1 -o yaml
apiVersion: v1
kind: Node
metadata:
...
spec:
  podCIDR: 10.11.0.0/24
  podCIDRs:
  - 10.11.0.0/24
  - fd04::/112
...

vagrant@k8s1:~/go/src/github.com/cilium/cilium$ kubectl get ciliumnodes.cilium.io -o yaml
apiVersion: v1
items:
- apiVersion: cilium.io/v2
  kind: CiliumNode
  ...
  spec:
    ...
    ipam:
      podCIDRs:
      - 10.11.0.0/24
      - fd04::/112
...

Fixes: #19594

Signed-off-by: Roman Ptitcyn <romanspb@yahoo.com>

julianwiedmann reviewed (approved):
LGTM, thanks @romanspb80 !

pchaigno reviewed (approved):
I've tested this locally.
